### PR TITLE
[Player-5077] Create/edit Picture in picture sample app to show & test behavior

### PR DESCRIPTION
**Pull request by oSulyma**
SDK contains API for using PIP in React-native-app via **PictureInPictureSampleApp**. It not a final changes, just for show & test behavior. All connected tickets should be merged into epic-4933.

Branch has one known issue, so PLEASE, don't merge it into dev, just review code. 
Issue is: **OoyalaSkinSampleApp** can't compile project from this state of **OoyalaSkinSDK.framework**, because of lack of access to required for PIP API in class `OOOoyalaPlayer+Playback.h` (that is part of **OOyalaSDK.framework**)

### CODE REVIEW

File: sdk/react/panels/AudioAndCCSelectionPanel.js [hunk 1]
```diff
@@ -58,15 +58,8 @@ class AudioAndCCSelectionPanel extends Component {
...
-    const localizedTitleForUndefinedLanguage = Utils.localizedString(this.props.config.locale, stringConstants.undefinedLanguageTitle, this.props.config.localizableStrings);
-    const localizedTitleForNoLinguisticContent = Utils.localizedString(this.props.config.locale, stringConstants.noLinguisticContentTitle, this.props.config.localizableStrings);
-    let originalName = name;
-
-    originalName = originalName.replace(localizedTitleForUndefinedLanguage, stringConstants.undefinedLanguageTitle);
-    originalName = originalName.replace(localizedTitleForNoLinguisticContent, stringConstants.noLinguisticContentTitle);
-
-    if (this.props.selectedAudioTrackTitle !== originalName) {
-      this.props.onSelectAudioTrack(originalName);
+    if (this.props.selectedAudioTrackTitle !== name) {
```
Comment (loginov-rocks) [inline sdk/react/panels/AudioAndCCSelectionPanel.js]: Was it an intentional change?

File: sdk/react/panels/AudioAndCCSelectionPanel.js [hunk 2]
```diff
@@ -150,8 +143,8 @@ class AudioAndCCSelectionPanel extends Component {
     return (
       <ItemSelectionScrollView
         style={styles.panelItemSelectionView}
-        items={itemsWithLocalizedUndefinedLanguage}
-        selectedItem={selectedLocalizedItem}
+        items={this.props.audioTracksTitles}
```
Comment (loginov-rocks) [inline sdk/react/panels/AudioAndCCSelectionPanel.js]: Was it an intentional change? I mean it looks out of PIP scope, right?

**Review by loginov-rocks — CHANGES_REQUESTED** [review]

File: sdk/react/ooyalaSkinBridgeListener.js [hunk 3]
```diff
@@ -338,6 +340,19 @@ class OoyalaSkinBridgeListener {
     this.core.handleControlsTouch();
   };
 
+  onPipToggle(e) {
+    Log.log("Received PiP Toggle: " + e.isPipActivated + e.isPipButtonVisible);
+    this.skin.setState({
+      isRootPipActivated: e.isPipActivated,
+      //isRootPipButtonVisible: e.isPipButtonVisible
```
Comment (ooyala-amohnatkin) [inline sdk/react/ooyalaSkinBridgeListener.js]: please remove the comment

File: sdk/react/ooyalaSkinBridgeListener.js [hunk 4]
```diff
@@ -338,6 +340,19 @@ class OoyalaSkinBridgeListener {
...
+    });
+    if (e.isPipButtonVisible) {
+      this.skin.setState({
```
Comment (ooyala-amohnatkin) [inline sdk/react/ooyalaSkinBridgeListener.js]: each `setState` can call render, so it's better to do several changes at one moment

File: sdk/react/ooyalaSkinBridgeListener.js [hunk 5]
```diff
@@ -338,6 +340,19 @@ class OoyalaSkinBridgeListener {
     this.core.handleControlsTouch();
   };
 
+  onPipToggle(e) {
+    Log.log("Received PiP Toggle: " + e.isPipActivated + e.isPipButtonVisible);
+    this.skin.setState({
+      isRootPipActivated: e.isPipActivated,
+      //isRootPipButtonVisible: e.isPipButtonVisible
+    });
+    if (e.isPipButtonVisible) {
```
Comment (ooyala-amohnatkin) [inline sdk/react/ooyalaSkinBridgeListener.js]: it means that `isRootPipButtonVisible` will never be set to `false`

File: sdk/react/panels/AudioAndCCSelectionPanel.js (hunk 1 above)
Comment (ooyala-amohnatkin) [inline sdk/react/panels/AudioAndCCSelectionPanel.js]: Why there are so many changes in audio tracks?

File: sdk/react/widgets/OOPiPView.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+import PropTypes from 'prop-types';
+
+import React, { Component } from 'react';
+import {
+  View, 
+  requireNativeComponent, 
+} from 'react-native';
+
+const iface = {
```
Comment (ooyala-amohnatkin) [inline sdk/react/widgets/OOPiPView.js]: it's not used anywhere, please clean up this file

File: sdk/react/panels/AudioAndCCSelectionPanel.js (hunk 1 above)
Comment (oSulyma) [reply sdk/react/panels/AudioAndCCSelectionPanel.js]: No, probable it was inherited from `dev`-branch

File: sdk/react/panels/AudioAndCCSelectionPanel.js (hunk 2 above)
Comment (oSulyma) [reply sdk/react/panels/AudioAndCCSelectionPanel.js]: No, it's not my changes and I don't use this snippet. Probable it was inherited from `dev`-branch .

File: sdk/react/panels/AudioAndCCSelectionPanel.js (hunk 1 above)
Comment (oSulyma) [reply sdk/react/panels/AudioAndCCSelectionPanel.js]: It's not my changes and I don't use this snippet. Probable it was inherited from `dev`-branch .

File: sdk/react/ooyalaSkinBridgeListener.js (hunk 5 above)
Comment (oSulyma) [reply sdk/react/ooyalaSkinBridgeListener.js]: Thank you. Such logic of handling if similar to Obj-C, but differs from Swift (that I relied on).
I decided that visibility is not necessary for this event (`onPipToggle`) and deleted it.

File: sdk/react/ooyalaSkinBridgeListener.js (hunk 4 above)
Comment (oSulyma) [reply sdk/react/ooyalaSkinBridgeListener.js]: I decided that visibility is not necessary for this event (onPipToggle) and deleted it.

**Review by maxkupetskii — APPROVED** [review]

**Review by loginov-rocks — APPROVED** [review]
Good work!

**Review by ooyala-amohnatkin — APPROVED** [review]

**Pull request merged**